pip/req/req_set.py

```diff
@@ -570,13 +602,32 @@ def _pip_has_created_build_dir(self):
             )
         )
 
+    def _to_install(self):
+        """Create the installation order.
+
+        We install the user specified things in the order given, except when
+        dependencies require putting other things first.
```

Comment:
That issue asked for this for that reason, but I mentioned in that issue if it wasn't for the fact that we should install dependencies first before the things that depend on them anyways, that I would have closed that issue because this isn't what ``install_requires`` is for.